R/carehomes.R

```diff
@@ -817,10 +796,207 @@ carehomes_severity <- function(p) {
+##'
+##' @param p_ICU A list with `date` and `value` for changes in p_ICU (the
+##'   probability of an hospitalised individual going to ICU). If
+##'   `value` is `NULL` then the value from `severity` is used. If `value` is
+##'   scalar then `date` can be `NULL` or unspecified. If `value` is a vector
+##'   then `date` must be a vector of sircovid dates of the same length as
+##'   `value`.
+##'
+##' @param p_H_D A list with `date` and `value` for changes in p_H_D (the
+##'   probability of death in general beds). If `value` is `NULL` then the value
+##'   from `severity` is used. If `value` is scalar then `date` can be `NULL` or
+##'   unspecified. If `value` is a vector then `date` must be a vector of
+##'   sircovid dates of the same length as `value`.
+##'
+##' @param p_ICU_D A list with `date` and `value` for changes in p_ICU_D (the
+##'   probability of death in ICU). If `value` is `NULL` then the value from
+##'   `severity` is used. If `value` is scalar then `date` can be `NULL` or
+##'   unspecified. If `value` is a vector then `date` must be a vector of
+##'   sircovid dates of the same length as `value`.
+##'
+##' @param p_W_D A list with `date` and `value` for changes in p_W_D (the
+##'   probability of death in stepdown). If `value` is `NULL` then the value
+##'   from `severity` is used. If `value` is scalar then `date` can be `NULL` or
+##'   unspecified. If `value` is a vector then `date` must be a vector of
+##'   sircovid dates of the same length as `value`.
+##'
+##' @param p_G_D A list with `date` and `value` for changes in p_G_D (the
+##'   probability of individuals requiring hospitalisation dying in the
+##'   community or a care home). If `value` is `NULL` then the value
+##'   from `severity` is used. If `value` is scalar then `date` can be `NULL` or
+##'   unspecified. If `value` is a vector then `date` must be a vector of
+##'   sircovid dates of the same length as `value`.
+##'
+##' @param p_G_D_CHR A list with `date` and `value` for changes in p_G_D (the
+##'   probability of individuals requiring hospitalisation dying in a care home)
+##'   for care home residents. If `value` is `NULL` then the value for the
+##'   oldest age group is used. If `value` is scalar then `date` can be `NULL`
+##'   or unspecified. If `value` is a vector then `date` must be a vector of
+##'   sircovid dates of the same length as `value`.
+##'
+##' @param p_star A list with `date` and `value` for changes in p_star (the
+##'   probability of patients being confirmed as covid on admission to
+##'   hospital). If `value` is `NULL` then the value
+##'   from `severity` is used. If `value` is scalar then `date` can be `NULL` or
+##'   unspecified. If `value` is a vector then `date` must be a vector of
+##'   sircovid dates of the same length as `value`.
+##'
+##' @return A list of severity parameters
+##'
+##' @export
+carehomes_parameters_severity <- function(dt,
+                                          severity = NULL,
```

Comment:
Generally, I think you could write this with defaults for `p_C` etc as `NULL` which will make this easier to scan.

Reply:
These are all now NULL as default (rather than a list of NULLs)